src/core/exception.d

```diff
@@ -416,12 +427,48 @@ deprecated void setAssertHandler( AssertHandler h ) @trusted nothrow @nogc
     assertHandler = h;
 }
 
+/*
+Returns a singleton object of class type `C`, i.e. repeated calls to
+`singleton!C` within a given thread return the same instance. For that reason,
+no constructor parameters are accepted; `C` must allow default construction.
+
+Different threads own different instances of `C`.
+
+The destructor of the singleton object is called during application's exit.
+*/
+private C singleton(C)() if (is(C == class) && is(typeof(new C)))
+{
+    static size_t[1 + (__traits(classInstanceSize, C) - 1) / size_t.sizeof] b;
+    static bool initialized = false;
+    auto result = () @trusted { return cast(C) b.ptr; }();
+    if (!initialized)
```

Comment:
I'm glad you're giving the same advice I often do :). I'd say when the if is short things are fine either way.